src/main/java/net/snowflake/client/core/HttpClientSettingsKey.java

```diff
@@ -132,8 +131,8 @@ public String getNonProxyHosts() {
     return this.nonProxyHosts;
   }
 
-  public Protocol getProxyProtocol() {
-    return this.proxyProtocol.equalsIgnoreCase("https") ? Protocol.HTTPS : Protocol.HTTP;
+  public String getProxyProtocol() {
```

Comment:
let's introduce own `Protocol` enum and use it instead of the `com.amazonaws.Protocol` - enums are better than strings when we have only small possible set of values